app/services/host_counter.rb

```diff
@@ -3,26 +3,39 @@ def initialize(association)
     @association = association
   end
 
+  delegate :fetch, to: :hosts_count
+
   def [](instance)
-    counted_hosts[instance&.id] || 0
+    hosts_count[instance&.id] || 0
   end
 
   def hosts_count
-    counted_hosts
+    # Use caching to save recounts in case new counters are created in
+    # quick succession, e.g. from CSV export or child hostgroup counts
+    cache do
+      counted_hosts
+    end
   end
 
   private
 
+  def cache
+    delay = Rails.env.test? ? 0 : 2.minutes
```

Comment:
Don't we use /dev/null cache for tests anyway?

Reply:
We dont. We should use "write anything, read nothing" cache for most of our codebase. I believe this is source of random failures. Only those test which need cache itself should explicitly enable that